Add scenarios to the tutorials (#3702)
wso2/docs-apim: en/docs/tutorials/scenarios/scenario12-message-delivery.md

# Scenario 12 - Guaranteed Message Delivery

## User Story

**Railco wants to partner up with a Catering service to provide food to their first class passengers. 

> For each registration they want to send an entry to the Catering service system. But they have noticed that 

Reviewer: ```suggestion
For each registration they want to send an entry to the Catering service system. However, they have noticed that 
```

If you look at the docker-compose.yml file, you can see that ``CATERING_SERVICE_EP`` is defined as 
‘<code>[http://www.urldoesnotexist.com](http://www.urldoesnotexist.com)</code>’ under mi-runtime service. 
If you send a request to this endpoint, it will fail and the message will be stored in the Message Store database.

To invoke the API, you can use the following curl.

```
curl -X POST 'http://localhost:8290/catering/register' \
--header 'Content-Type: application/json' \
--data-raw '{
 "passengerId": "241241d",
 "class": "first",
 "train": "RLC-5051",
 "departure": {
   "time": "2021-07-16 11:45",
   "station": "King Cross"
 },
 "arrival": {
   "time": "2021-07-16 16:40",
   "station": "Edinburgh"
 },
 "duration": "4h 55m",
 "food": [
   {
     "serving": "lunch",
     "type": "Asian",
     "comments": [
       "Not too spicy"
     ]
   },
   {
     "serving": "snacks",
     "type": "Normal",
     "comments": []
   }
 ]
}'

```

If you check the Micro Integrator logs, you can see that the Backend invocation has failed.

“Message processor [CateringServiceForwardingProcessor] failed to forward message 4 times. 
Moved failed message to fail-messages-store and continue”

> After sometime, the Micro Integrator will again try to send the message to the Backend. 

Reviewer: ```suggestion
After some time, the Micro Integrator will again try to send the message to the Backend. 
```

This process will continue until the message is successfully delivered to the backend.

To make the delivery successful, you can provide a working endpoint and restart the Micro Integrator. 
This will send the above message to that backend. 

> To test you can create a new endpoint in [https://hookbin.com/](https://hookbin.com/) and provide 

Reviewer: ```suggestion
To test this flow you can create a new endpoint in [https://hookbin.com/](https://hookbin.com/) and provide 
```
